cmake/CMakeLists.txt

```diff
@@ -249,6 +249,33 @@ if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
   endif()
 endif()
 
+#######################################
+# add custom target for IWYU analysis
+#######################################
+set(ENABLE_IWYU OFF CACHE BOOL "Add 'iwyu' build target to call the include-what-you-use tool")
+mark_as_advanced(ENABLE_IWYU)
+if(ENABLE_IWYU)
+  find_program(IWYU_EXE NAMES include-what-you-use iwyu)
+  find_program(IWYU_TOOL NAMES iwyu_tool iwyu-tool iwyu_tool.py)
+  if (IWYU_EXE AND IWYU_TOOL)
+    add_custom_target(
+      iwyu
+      ${IWYU_TOOL} -o clang -p ${CMAKE_CURRENT_BINARY_DIR} -- -Xiwyu --mapping_file=${CMAKE_CURRENT_BINARY_DIR}/iwyu-extra-map.imp
+      COMMENT "Running IWYU")
+    add_dependencies(iwyu lammps)
+    # generate file with additional mappings
+    file(WRITE  ${CMAKE_CURRENT_BINARY_DIR}/iwyu-extra-map.imp "[\n")
```

Comment:
All the file content is static, so is there a reason to generate this? It just seems nicer to put that file somewhere in `cmake/` .

Reply:
this was done due to a combination of laziness and ignorance. I just used the first thing that came to my mind that kept everything in place since this PR is touching a lit of file (now nearly 1000).

I am happy to apply an update to this, if you propose something simple (once I am done with the updates to the sources).

Reply:
https://github.com/akohlmey/lammps/pull/66